reasoner/controller/ConjunctionController.java

```diff
@@ -184,80 +194,115 @@ NegatedRequest asNegated() {
+        final Set<Variable.Retrievable> outputVariables;
+        private final Map<ConjunctionStreamPlan, Map<ConceptMap, PoolingStream.BufferedFanStream<ConceptMap>>> compoundStreamRegistry;
 
         Processor(Driver<PROCESSOR> driver,
                   Driver<? extends ConjunctionController<?, PROCESSOR>> controller,
-                  Context context, ConceptMap bounds, List<Resolvable<?>> plan,
+                  Context context, ConceptMap bounds, ConjunctionStreamPlan conjunctionStreamPlan,
                   Supplier<String> debugName) {
             super(driver, controller, context, debugName);
             this.bounds = bounds;
-            this.plan = plan;
+            this.outputVariables = controller.actor().outputVariables;
+            this.plan = conjunctionStreamPlan;
+            this.compoundStreamRegistry = new HashMap<>();
             context.perfCounters().conjunctionProcessors.add(1);
         }
 
         public class CompoundStream extends TransformationStream<ConceptMap, ConceptMap> {
-
-            private final Publisher<ConceptMap> leadingPublisher;
-            private final List<Resolvable<?>> remainingPlan;
-            private final Map<Publisher<ConceptMap>, ConceptMap> publisherPackets;
-            private final ConceptMap initialPacket;
+            private final ConjunctionStreamPlan plan;
+            private final ConceptMap identifierBounds;
             private final AbstractProcessor<?, ?, ?, ?> processor;
+            private final Map<Publisher<ConceptMap>, Integer> whichChild;
 
-            CompoundStream(AbstractProcessor<?, ?, ?, ?> processor, List<Resolvable<?>> plan,
-                           ConceptMap initialPacket) {
-                super(processor, new SubscriberRegistry.Single<>(), new PublisherRegistry.Multi<>());
+
+            CompoundStream(AbstractProcessor<?, ?, ?, ?> processor, ConjunctionStreamPlan plan, ConceptMap identifierBounds) {
+                super(processor, new SubscriberRegistry.Multi<>(), new PublisherRegistry.Multi<>());
                 this.processor = processor;
-                assert plan.size() > 0;
-                this.initialPacket = initialPacket;
-                this.remainingPlan = new ArrayList<>(plan);
-                this.publisherPackets = new HashMap<>();
-                this.leadingPublisher = nextCompoundLeader(this.remainingPlan.remove(0), initialPacket);
-                this.leadingPublisher.registerSubscriber(this);
+                this.identifierBounds = identifierBounds;
+                this.plan = plan;
+                Publisher<ConceptMap> leftChild;
+                if (plan.isCompoundStream()) {
+                    leftChild = spawnPlanElement(plan.asCompoundStreamPlan().ithChild(0), identifierBounds);
+                } else {
+                    leftChild = spawnPlanElement(plan.asResolvablePlan(), identifierBounds);
+                }
+                leftChild.registerSubscriber(this);
```

Comment:
Creates the first child based on input bounds.
Remaining children are created when children on their left (before them in the plan) return an answer.